extensions/mssql/src/types.ts

```diff
@@ -24,3 +24,25 @@ export function isUndefinedOrNull(obj: any): boolean {
 export function isUndefined(obj: any): boolean {
 	return typeof (obj) === _typeof.undefined;
 }
+
+
+/**
+ * Deferred promise
+ */
+export class Deferred<T = void> {
```

Comment:
Why aren't you using https://github.com/microsoft/vscode-mssql/blob/main/src/protocol.ts#L31?

Reply:
how do I use that? that's in the extension and this is ADS